src/Eventso.Subscription.Kafka.DeadLetter.Postgres/PoisonEventStore.cs

```diff
@@ -0,0 +1,386 @@
+                offsetParameter.TypedValue = poisonEvent.TopicPartitionOffset.Offset.Value;
+                keyParameter.TypedValue = poisonEvent.Key;
+                valueParameter.TypedValue = poisonEvent.Value;
+                creationTimestampParameter.TypedValue = poisonEvent.CreationTimestamp;
+                headerKeysParameter.TypedValue = headerKeys;
+                headerValuesParameter.TypedValue = headerValues;
+                lastFailureReasonParameter.TypedValue = poisonEvent.FailureReason;
+
+                await command.ExecuteNonQueryAsync(token);
+            }
+        }
+
+        public async Task AddFailures(
+            DateTime timestamp,
+            IReadOnlyCollection<OccuredFailure> failures,
+            CancellationToken token)
+        {
+            var topics = new String[failures.Count];
+            var partitions = new int[failures.Count];
+            var offsets = new long[failures.Count];
+            var reasons = new string[failures.Count];
+            foreach (var (index, failure) in Enumerate(failures))
+            {
+                topics[index] = failure.TopicPartitionOffset.Topic;
+                partitions[index] = failure.TopicPartitionOffset.Partition.Value;
+                offsets[index] = failure.TopicPartitionOffset.Offset.Value;
+                reasons[index] = failure.Reason;
+            }
+            
+            await using var connection = _connectionFactory.ReadWrite();
+
+            await using var command = new NpgsqlCommand(
+                @"
+UPDATE eventso_dlq.poison_events pe
+SET
+    last_failure_timestamp = @timestamp,
+    last_failure_reason = input.reason,
+    total_failure_count = total_failure_count + 1
+FROM UNNEST(@topics, @partitions, @offsets, @reasons) AS input(topic, partition, ""offset"", reason)
+WHERE pe.topic = input.topic AND pe.partition = input.partition AND pe.""offset"" = input.""offset"";",
+                connection)
+            {
+                Parameters =
+                {
+                    new NpgsqlParameter<string[]>("topics", topics),
+                    new NpgsqlParameter<int[]>("partitions", partitions),
+                    new NpgsqlParameter<long[]>("offsets", offsets),
+                    new NpgsqlParameter<DateTime>("timestamp", timestamp),
+                    new NpgsqlParameter<string[]>("reasons", reasons)
+                }
+            };
+
+            await connection.OpenAsync(token);
+
+            await command.ExecuteNonQueryAsync(token);
+        }
+
+        public async Task Remove(IReadOnlyCollection<TopicPartitionOffset> topicPartitionOffsets, CancellationToken token)
+        {
+            var topics = new string[topicPartitionOffsets.Count];
+            var partitions = new int[topicPartitionOffsets.Count];
+            var offsets = new long[topicPartitionOffsets.Count];
+            foreach (var (index, topicPartitionOffset) in Enumerate(topicPartitionOffsets))
```

Comment:
look like the same as foreach (var (index, topicPartitionOffset) in topicPartitionOffsets.Select(x, i) => (i,x))

Reply:
++